polkadot/runtime/parachains/src/scheduler/tests.rs

```diff
@@ -186,7 +189,10 @@ fn claimqueue_ttl_drop_fn_works() {
 		assert!(!claimqueue_contains_para_ids::<Test>(vec![para_id]));
 
 		// Add a claim on core 0 with a ttl in the future (15).
-		let paras_entry = ParasEntry::new(Assignment::new(para_id), now + 5);
+		let paras_entry = ParasEntry::new(
+			UnifiedAssignmentType::<Test>::OnDemand(OnDemandAssignment::new(para_id, core_idx)),
```

Comment:
We could and likely should use a test config that uses a mock test assignment provider, so we don't need to expose internals of the used assignment provider here.

Reply:
Not too familiar with how I'd make a mock test assignment provider. If my above suggestion can substitute, that would probably be easier for me to implement.